oci/oci.go

```diff
@@ -62,6 +47,67 @@ const (
 	UntrustedRuntime = "untrusted"
 )
 
+// Runtime is the generic structure holding both global and specific
+// information about the runtime through RuntimeBase and RuntimeImpl
+// respectively.
+type Runtime struct {
+	RuntimeBase
+	RuntimeImpl
+}
+
+// RuntimeBase gathers informations that can be shared across Runtime
+// implementations.
+type RuntimeBase struct {
+	name                     string
+	trustedPath              string
+	untrustedPath            string
+	trustLevel               string
+	runtimes                 map[string]RuntimeHandler
+	conmonPath               string
+	conmonEnv                []string
+	cgroupManager            string
+	containerExitsDir        string
+	containerAttachSocketDir string
+	logSizeMax               int64
+	noPivot                  bool
+	ctrStopTimeout           int64
+}
+
+// RuntimeImpl is an interface used by the caller to interact with the
+// container runtime. The purpose of this interface being to abstract
+// implementations and their associated assumptions regarding the way to
+// interact with containers. This will allow for new implementations of
+// this interface, especially useful for the case of VM based container
+// runtimes. Assumptions based on the fact that a container process runs
+// on the host will be limited to the RuntimeV1 implementation.
+type RuntimeImpl interface {
+	Version() (string, error)
+	CreateContainer(*Container, string) error
+	StartContainer(*Container) error
+	Exec(*Container, []string, io.Reader, io.WriteCloser, io.WriteCloser,
```

Comment:
nit: We should call this `ExecContainer` for uniformity. (This can be done in a follow-up.)